core/src/main/java/org/modelmapper/internal/converter/MapConverter.java

```diff
@@ -80,8 +82,12 @@ public MatchResult match(Class<?> sourceType, Class<?> destinationType) {
 
   protected Map<Object, Object> createDestination(
       MappingContext<Map<?, ?>, Map<Object, Object>> context) {
-    if (context.getDestinationType().isInterface())
+    if (context.getDestinationType().isInterface()) {
+      if (SortedMap.class.isAssignableFrom(context.getDestinationType())) {
+        return new TreeMap<Object, Object>();
+      }
       return new HashMap<Object, Object>();
+    }
 
     return context.getMappingEngine().createDestination(context);

```

Comment:
What do you think:

```java
if (!context.getDestinationType().isInterface())
  return context.getMappingEngine().createDestination(context);
if (SortedMap.class.isAssignableFrom(context.getDestinationType()))
  return new TreeMap<Object, Object>();
return new HashMap<Object, Object>();
```

to reduce the deeply nested if statement.

And we prefer no curly braces if there is only one statement in the if/else as our coding stye.

Reply:
For me, your variant is less readable but it's my IMHO. ;) Codestyle is totally your call.
I'll update the PR.